### DISCUSSION TITLE
If statement in parsed function

### DISCUSSION THREAD

**Discussion by steam-simulator** [idaholab/moose#23176] [Q&A General] [answered]
Hi,
I’m trying to use an ‘if’ statement to define the following statements in a parsed function:

if t < 30, then 10  
elseif t is between 30 and 40, then 5 
elseif  t is between 40 and 60, then 10
else 5.

I tried using the following, but it doesn’t work. Please, help.
`value = 'if(t <= 30, 10, if(t <= 40, 5.0),  if(t <= 60, 10.0, 5.0))'`

**Comment by ajacquey** [comment, marked as answer]
The `if` command always take 3 arguments. Try this:

`value = 'if(t <= 30, 10, if(t <= 40, 5.0,  if(t <= 60, 10.0, 5.0)))'`

**Reply by steam-simulator** [reply]
That works great! Thanks, @ajacquey

**Comment by Hejie-Guo** [comment]
can this parsed function also be expressed in that way ?
![image](https://github.com/idaholab/moose/assets/61571563/45f200f4-e6f1-414d-a7fe-db497d2815f9)

**Reply by GiudGiud** [reply]
hello

Yes, you can nest if conditions to achieve this
```
expression = 'if(t <= 1500, if(x <= 1500, 900,  0), if(x <= 1500, 100.0, if(x < 2000, 200, 300))))'

```

**Reply by Hejie-Guo** [reply]
Thank you very much. Did there any files tell us how to use this ParsedFunction more details?

**Reply by GiudGiud** [reply]
Hello

There's the documentation for fparser which we rely on in the backend
http://warp.povusers.org/FunctionParser/
